frontend/src/PagesAdmin/SultenMenuItemFormAdminPage/SultenMenuItemFormAdminPage.tsx

```diff
@@ -0,0 +1,209 @@
+import { DropDownOption } from '~/Components/Dropdown/Dropdown';
+import { dbT, lowerCapitalize } from '~/utils';
+import { AdminPageLayout } from '../AdminPageLayout/AdminPageLayout';
+
+export function SultenMenuItemFormAdminPage() {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+
+  // Form data
+  const { id } = useParams();
+  const [showSpinner, setShowSpinner] = useState<boolean>(true);
+  const [menuItem, setMenuItem] = useState<Partial<MenuItemDto>>({});
+  const [foodPreferenceOptions, setFoodPreferenceOptions] = useState<DropDownOption<number>[]>([]);
+  const [foodCategoryOptions, setFoodCategoryOptions] = useState<DropDownOption<number>[]>([]);
+  // Fetch data if edit mode.
+
+  useEffect(() => {
+    Promise.all([
+      getFoodCategorys()
+        .then((data) => {
+          setFoodCategoryOptions(
+            data.map(
+              (category: FoodCategoryDto) =>
+                ({
+                  label: dbT(category, 'name'),
+                  value: category.id,
+                }) as DropDownOption<number>,
+            ),
+          );
+        })
+        .catch((error) => {
+          toast.error(t(KEY.common_something_went_wrong));
+          console.error(error);
+        }),
+      getFoodPreferences()
+        .then((data) => {
+          setFoodPreferenceOptions(
+            data.map(
+              (preference: FoodPreferenceDto) =>
+                ({
+                  label: dbT(preference, 'name'),
+                  value: preference.id,
+                }) as DropDownOption<number>,
+            ),
+          );
+        })
+        .catch((error) => {
+          toast.error(t(KEY.common_something_went_wrong));
+          console.error(error);
+        }),
+    ]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (id) {
+      getMenuItem(id)
+        .then((data) => {
+          setMenuItem(data);
+          setShowSpinner(false);
+        })
+        .catch((data) => {
+          // TODO add error pop up message?
+          if (data.request.status === STATUS.HTTP_404_NOT_FOUND) {
+            navigate(ROUTES.frontend.admin_recruitment);
+          }
+          toast.error(t(KEY.common_something_went_wrong));
+          console.error(data);
+        });
+    } else {
+      setShowSpinner(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id]);
+
+  const initialData: Partial<MenuItemDto> = {
+    name_nb: menuItem?.name_nb,
+    name_en: menuItem?.name_en,
+
+    description_nb: menuItem?.description_nb,
+    description_en: menuItem?.description_en,
+
+    price: menuItem?.price,
+    price_member: menuItem?.price_member,
+
+    food_preferences: [], // TODO add on multiselect
+    food_category: (menuItem?.food_category as FoodCategoryDto)?.id,
+  };
+
+  const submitText = id ? t(KEY.common_save) : t(KEY.common_create);
+
+  // Loading.
```

Comment:
Du kan legge dette rett under effects
```suggestion
  // Guards.
```

Reply:
Done